tests/js/server/upgrade-data/upgrade-data-collection-upgrade-msv3.6.0.js

```diff
@@ -0,0 +1,163 @@
+/*jshint globalstrict:false, strict:false */
+/*global arango, assertEqual, assertTrue, assertEqual, assertTypeOf, assertNotEqual, fail */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test the view interface
+///
+/// @file
+///
+/// DISCLAIMER
+///
+/// Copyright 2010-2012 triagens GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Dan Larkin-York
+/// @author Copyright 2012, triAGENS GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+var jsunity = require("jsunity");
+var db = require('internal').db;
+const request = require('@arangodb/request');
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite: upgrade with data from mmfiles instance
+////////////////////////////////////////////////////////////////////////////////
+
+function UpgradeData () {
+  'use strict';
+
+  const verifyCollectionData = () => {
+    const c = db._collection('CollectionUpgrade');
+    assertEqual(c.count(), 100);
+
+    // verify documents and contents
+    for (let i = 0; i < 100; i++) {
+      const doc = c.document({ _key: `key${i}` });
+      assertEqual(doc.name, `Name ${i}`);
+      assertEqual(doc.num, i);
+      assertEqual(doc.num10, i % 10);
+      assertEqual(doc.location, { coordinates: [i, i], type: 'Point' });
+    }
+
+    // verify indexes
+    const indices = c.getIndexes();
+    assertEqual(indices.length, 5);
+
+    // primary
```

Comment:
I think that upgrading edge indexes has an issue (please refer to the `prefix_same_as_start` comments above), so we should add tests here for an edge collection as well!

Reply:
I will add a test. Need to generate new data for the upgrade-data-tests repo first...